### PR TITLE
Shared canvas for volmeters

### PR DESCRIPTION
**Pull request by holiber**
Reverts stream-labs/streamlabs-obs#2660
Fixed the bug that prevented FPS capping
We can manually setup FPS limit for volmeters in experimental settings
image

### CODE REVIEW

**Comment by andycreeth** [comment]
I remember there was a bug from last time we reverted this PR that looks like it hasn't been fixed.  If the name of the source overflows onto the next line it overlaps the volmeter:

image

**Comment by holiber** [comment]
Should be fixed now 
image

**Review by andycreeth — CHANGES_REQUESTED** [review]
Good stuff.  Have a couple minor comments.

File: app/components/editor/elements/BaseElement.tsx [hunk 1]
```diff
@@ -11,20 +11,34 @@ export default class BaseElement extends TsxComponent {
...
 
-  mounted() {
-    this.setDimensions();
-    window.setInterval(() => this.setDimensions(), 500);
+  static sizeWatcherId: number;
+  static sizeWatcherCallbacks: Function[] = [];
+  static addSizeWatcher(cb: Function) {
+    this.sizeWatcherCallbacks.push(cb);
+    if (this.sizeWatcherId) return;
+    this.sizeWatcherId = window.setInterval(() => {
```
[the remark this answers is not in the document]
Comment (holiber) [reply app/components/editor/elements/BaseElement.tsx:21]: I don't think we need to clear it. It's only one 1 interval per app launch.

**Review by andycreeth — APPROVED** [review]

**Pull request merged**